development/mimir-microservices-mode/compose-up.sh

```diff
@@ -6,13 +6,22 @@
 
 set -e
 
-SCRIPT_DIR=$(cd "$(dirname "$0")" && pwd)
-BUILD_IMAGE=$(make -s -f ${SCRIPT_DIR}/../../Makefile print-build-image)
+# newer compose is a subcommand of `docker`, not a hyphenated standalone command
+docker_compose() {
+    if [ -x "$(command -v docker-compose)" ]; then
+        docker-compose "$@"
+    else
+        docker compose "$@"
+    fi
+}
+
+SCRIPT_DIR=$(cd "$(dirname -- "$0")" && pwd)
+BUILD_IMAGE=$(make -s -f "${SCRIPT_DIR}"/../../Makefile print-build-image)
 # Make sure docker-compose.yml is up-to-date.
-cd $SCRIPT_DIR && make
+cd "$SCRIPT_DIR" && make
 
 # -gcflags "all=-N -l" disables optimizations that allow for better run with combination with Delve debugger.
 # GOARCH is not changed.
-CGO_ENABLED=0 GOOS=linux go build -mod=vendor -gcflags "all=-N -l" -o ${SCRIPT_DIR}/mimir ${SCRIPT_DIR}/../../cmd/mimir
-docker-compose -f ${SCRIPT_DIR}/docker-compose.yml build --build-arg BUILD_IMAGE=${BUILD_IMAGE} distributor-1
-docker-compose -f ${SCRIPT_DIR}/docker-compose.yml up $@
+CGO_ENABLED=0 GOOS=linux go build -mod=vendor -gcflags "all=-N -l" -o "${SCRIPT_DIR}"/mimir "${SCRIPT_DIR}"/../../cmd/mimir
+docker_compose -f "${SCRIPT_DIR}"/docker-compose.yml build --build-arg BUILD_IMAGE="${BUILD_IMAGE}" distributor-1
+docker_compose -f "${SCRIPT_DIR}"/docker-compose.yml up "$@"
```

Comment:
I don't think quotes around `$@` are correct -- that will convert all remaining argument into single argument.

Reply:
So it does work with multiple args, e.g.

`./compose-up.sh -d minio memcached distributor-1 distributor-2 ingester-1 ingester-2 store-gateway-1 store-gateway-2 querier query-scheduler query-frontend nginx grafana`

The quoting suppresses this warning, which is about using globbing  
https://github.com/koalaman/shellcheck/wiki/SC2068:
```
Let's say you have four arguments: baz, foo bar, * and /*/*/*/*

"$@" will expand into exactly that: baz, foo bar, * and /*/*/*/*

$@ will expand into multiple other arguments: baz, foo, bar, file.txt, otherfile.jpg, and (eventually) a list of most files on the system
```

I could see that someone may want to use globbing with the docker-compose command?
It's probably fine to leave it unquoted, as it's not intended to be a live install script or something run on a production system.

Reply:
Let's keep as-is, until someone runs into a problem here.